cdm/core/src/main/java/ucar/nc2/ft2/coverage/HorizCoordSys.java

```diff
@@ -646,12 +654,37 @@ private List<LatLonPoint> calcLatLonBoundaryPointsFromProjection(int maxPointsIn
     List<LatLonPoint> latLonPoints = new LinkedList<>();
 
     for (ProjectionPoint projPoint : projPoints) {
-      latLonPoints.add(transform.getProjection().projToLatLon(projPoint));
+      final ProjectionPoint projPointInKm =
+          ProjectionPoint.create(convertToKm(projPoint.getX(), xAxis.units, xAxis.name),
+              convertToKm(projPoint.getY(), yAxis.units, yAxis.name));
+      latLonPoints.add(transform.getProjection().projToLatLon(projPointInKm));
     }
 
     return latLonPoints;
   }
 
+  // TODO is there a better place to handle units?
+  private static double convertToKm(double coordinate, String unit, String axisName) {
+    if (unit.equals("km") || unit.equals("kilometers")) {
+      return coordinate;
+    } else if (unit.equals("m") || unit.equals("meters")) {
+      return 0.001 * coordinate;
+    } else {
+      // TODO not sure what other units that may not need scaling could be here (e.g. radians)
```

Comment:
maybe for now just log that it's an "unrecognized unit"?

Reply:
added a log message in 4fb4ea69-- only made it "info" level because if the units are radians for example then there is no problem.